pkg/imagestream/imagestreamhasblob.go

```diff
@@ -37,6 +38,27 @@ func (is *imageStream) HasBlob(ctx context.Context, dgst digest.Digest, requireM
 		return found
 	}
 
+	if layers, err := is.imageStreamGetter.layers(); err == nil {
+		if _, ok := layers.Blobs[dgst.String()]; !ok {
+			return logFound(nil)
+		}
+		copied := make([]imageapiv1.ImageLayer, 0, len(layers.Blobs))
+		for name, blob := range layers.Blobs {
+			copied = append(copied, imageapiv1.ImageLayer{Name: name, LayerSize: blob.LayerSize, MediaType: blob.MediaType})
+		}
+		var firstImage string
+		for name := range layers.Layers {
+			firstImage = name
+			break
+		}
+		return logFound(&imageapiv1.Image{
```

Comment:
what is layers.Layers was empty?

Reply:
that means the image stream has no layers

Reply:
yeah but we shouldn't returning found in that case, right?

Reply:
if we don't have a layer we return on line 43 with nil, and the if block fails, and we return ErrBlobUnknown.